### ISSUE TITLE
Sony Visca Tally on ILME-FR7K

### ISSUE THREAD

**Issue by BarbeRousseDeVicto** [bitfocus/companion-module-sony-visca#57]
Hi !

If you can add to the Visca feature the Tally lamp control on each camera it would be great !

I would like to be able to turn on the tally on a specific cam when it is selected with the Black Magic Atem.

**Comment by BarbeRousseDeVicto** [comment]
Anyone know how I can set the Tally Light On ?

I know the command to send for RED is 8x 01 7E 01 0A 00 0p FF but in the datasheet in the comment section they mention this "p: 2=On, 3=Off" where do I put this info ?

**Comment by BarbeRousseDeVicto** [comment]
Okay so the Sony Doc specify the following: 

> Expiration time for an on status of the tally lamp The tally lamp is turned off if not receiving an On command from any controller for 15 seconds after receiving an On command of TALLY ON/OFF. If the tally lamp is lit by an input other than VISCA over IP, such as a tally input from the OPTION connector or a CGI command, the tally lamp will not turn off.

Do you think you can set something in the module to make it Still Alive by any kind of ping process ?

**Comment by TobyZipon** [comment]
What @BarbeRousseDeVicto said would be good.

I had to add a separate loop-trigger (which sends a "ON" command every 10s) which is triggered by the original Tally on trigger. And of course turn off the loop with the OFF-trigger. It works but I notice some delay and it just seems a bit ineffective.

**Comment by goseid** [comment]
If we set something in the module to keep the tally on by sending the command every 10 seconds I'm concerned it may cause unexpected behavior for people that control the camera by another means such as a second Companion instance, the web interface or the tally functionality on the PTZ controller or switcher. If one stream deck keeps telling the camera Tally should be on every 10 seconds and a second one tells it to turn off, it will keep coming back on because of the first instance.

Should we implement something like a LockTallyOn action that is a little more clear that its semi-permanent?

**Comment by nicksimoneau** [comment]
> If we set something in the module to keep the tally on by sending the command every 10 seconds I'm concerned it may cause unexpected behavior for people that control the camera by another means such as a second Companion instance, the web interface or the tally functionality on the PTZ controller or switcher. If one stream deck keeps telling the camera Tally should be on every 10 seconds and a second one tells it to turn off, it will keep coming back on because of the first instance.
> 
> Should we implement something like a LockTallyOn action that is a little more clear that its semi-permanent?

Yeah I totaly agree with that. The goal is really to get it lockOn but the only idea to achive that was to redo the commande since there is no other walkaround as I know.
